torchgan/losses/historical.py

```diff
@@ -0,0 +1,67 @@
+import torch
+from .loss import GeneratorLoss, DiscriminatorLoss
+from ..utils import reduce
+
+__all__ = ['HistoricalAverageGeneratorLoss', 'HistoricalAverageDiscriminatorLoss']
+
+class HistoricalAverageGeneratorLoss(GeneratorLoss):
+    # TODO(Aniket1998): Find a generalisation for multi agent GANs
+    def __init__(self, reduction='elementwise_mean', override_train_ops=None):
+        super(HistoricalAverageGeneratorLoss, self).__init__(reduction, override_train_ops)
+        self.timesteps = 0
+        self.sum_parameters = []
+
+    def train_ops(self, generator, optimizer_generator):
+        if self.override_train_ops is not None:
+            return self.override_train_ops(self, generator, optimizer_generator)
+        else:
+            if self.timesteps == 0:
+                for p in generator.parameters():
+                    self.sum_parameters.append(torch.Tensor(p))
+                self.timesteps += 1
+                return 0.0
+            else:
+                optimizer_generator.zero_grad()
+                loss = None
+                for i, p in enumerate(generator.parameters()):
+                    if loss is None:
+                        loss = torch.sum((p - self.sum_parameters[i]) ** 2)
+                    else:
+                        loss += torch.sum((p - self.sum_parameters[i]) ** 2)
+                    self.sum_parameters[i] += p
+                loss /= self.timesteps
```

Comment:
Is this what the paper suggests?
You should be taking the average of the past parameters,i.e `self.sum_parameters[i] / self.timesteps`. You might divide the loss by the length of `generator.parameters` for normalization purposes.

Reply:
Fixed in the new commit, was a typo I overlooked